include/bluetooth/hci.h

```diff
@@ -626,6 +626,86 @@ struct bt_hci_rp_read_bd_addr {
 	bt_addr_t bdaddr;
 } __packed;
 
+/* logic transport types used when reading codecs and controller delay */
+#define BT_HCI_LOGICAL_TRANSPORT_TYPE_BREDR_ACL 0
+#define BT_HCI_LOGICAL_TRANSPORT_TYPE_BREDR_SCO 1
+#define BT_HCI_LOGICAL_TRANSPORT_TYPE_LE_CIS    2
+#define BT_HCI_LOGICAL_TRANSPORT_TYPE_LE_BIS    3
+
+/* audio datapath directions */
+#define BT_HCI_DATAPATH_DIR_HOST_TO_CTLR 0
+#define BT_HCI_DATAPATH_DIR_CTLR_TO_HOST 1
+
+/* coding format assigned numbers, used for codec IDs */
+#define BT_HCI_CODING_FORMAT_ULAW_LOG    0x00
+#define BT_HCI_CODING_FORMAT_ALAW_LOG    0x01
+#define BT_HCI_CODING_FORMAT_CVSD        0x02
+#define BT_HCI_CODING_FORMAT_TRANSPARENT 0x03
+#define BT_HCI_CODING_FORMAT_LINEAR_PCM  0x04
+#define BT_HCI_CODING_FORMAT_MSBC        0x05
+#define BT_HCI_CODING_FORMAT_VS          0xFF
+
+
+#define BT_HCI_OP_READ_CODECS                   BT_OP(BT_OGF_INFO, 0x000b)
+struct bt_hci_std_codec_info {
+	uint8_t codec_id;
+} __packed;
+struct bt_hci_vs_codec_info {
+	uint16_t company_id;
+	uint16_t codec_id;
+} __packed;
+struct bt_hci_rp_read_codecs {
+	uint8_t status;
+	/* other fields filled in dynamically */
+} __packed;
+
+#define BT_HCI_OP_READ_CODECS_V2                BT_OP(BT_OGF_INFO, 0x000d)
+struct bt_hci_std_codec_info_v2 {
+	uint8_t codec_id;
+	uint8_t transport;
+} __packed;
+struct bt_hci_vs_codec_info_v2 {
+	uint16_t company_id;
+	uint16_t codec_id;
+	uint8_t transport;
+} __packed;
+struct bt_hci_rp_read_codecs_v2 {
+	uint8_t status;
+	/* other fields filled in dynamically */
+} __packed;
```

Comment:
```suggestion
struct bt_hci_std_codec_info_v2 {
	uint8_t codec_id;
	uint8_t transports; /* bitmap */
} __packed;
struct bt_hci_std_codecs_v2 {
	uint8_t num_supp_std_codecs_v2;
	struct bt_hci_std_codec_info_v2 std_info_v2[0];
} __packed;
struct bt_hci_vs_codec_info_v2 {
	uint16_t company_id;
	uint16_t codec_id;
	uint8_t transports; /* bitmap */
} __packed;
struct bt_hci_vs_codecs_v2 {
	uint8_t num_supp_vs_codecs_v2;
	struct bt_hci_vs_codec_info_v2 vs_info_v2[0];
} __packed;

struct bt_hci_rp_read_codecs_v2 {
	uint8_t status;
	/* other fields filled in dynamically */
	struct bt_hci_codecs_v2 codecs_v2[0];
} __packed;
```

Reply:
ad `struct bt_hci_codecs_v2 codecs_v2[0]`: Structs with incomplete array members are not allowed to be elements of arrays or members of other structs. Technically we do not use incomplete array types, but we use zero-length arrays in the same way so I think we should adhere to that aspect of the C standard. Having a `uint8 codecs [0];` is fine by me though.